### PR TITLE
IPCs now take damage from Slimes feeding on them

### PR DESCRIPTION
**Pull request by WenlockTheBritishHobo**
## What Does This PR Do

Makes it so IPCs take damage when a slime is feeding on them, preventing the feeding from being free nutrition. Also adds two comments to make finding the slime feeding easier.

## Why It's Good For The Game

Fixes https://github.com/ParadiseSS13/Paradise/issues/13325 as nutrition shouldn't be free. 

## Testing

Spawned in as a Drask Scientist, spawned a grey slime, took control of slime, fed on Drask, saw Drask take correct damage.

Spawned in as an IPC Scientist, took control of previously spawned slime, fed on IPC, saw IPC take correct damage.

Spawned in an Alien Sentinel, took control of previously spawned slime, fed on Sentinel, saw Sentinel take correct damage.

Spawned in as an Vox Scientist, took control of previously spawned slime, fed on Vox, saw Vox take correct damage.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  <hr>

## Changelog

:cl:
fix: Slimes now deal damage to IPCs when feeding on them.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/slime/slime_life.dm [hunk 1]
```diff
@@ -155,7 +155,7 @@
 	if(prob(30) && stat == CONSCIOUS)
 		adjustBruteLoss(-1)
 
-/mob/living/simple_animal/slime/proc/handle_feeding()
+/mob/living/simple_animal/slime/proc/handle_feeding() // This is where slime feeding is actually done.
```
Comment (1080pCat) [inline code/modules/mob/living/simple_animal/slime/slime_life.dm:158]:
```suggestion
// This is where slime feeding is handled.
/mob/living/simple_animal/slime/proc/handle_feeding()
```

File: code/modules/mob/living/simple_animal/slime/slime_life.dm [hunk 2]
```diff
@@ -174,10 +174,19 @@
 		Feedstop()
 		return
 
-	if(iscarbon(M))
+	if(iscarbon(M)) // This is where damage done by slime feeding is done.
```
Comment (1080pCat) [inline code/modules/mob/living/simple_animal/slime/slime_life.dm:177]:
```suggestion
// This is where damage dealt by slime feeding is handled.
	if(iscarbon(M))
```

**Review by 1080pCat — COMMENTED** [review]
marks and similar things should go above code blocks

**Review by CinnamonSnowball — COMMENTED** [review]
Have you considered simply making slimes not gain any nutrition if no damage was dealt? That method would keep the unique immunity to slimes IPCs have, as opposed to removing it and making them more similar to other species

**Comment by WenlockTheBritishHobo** [comment]
> Have you considered simply making slimes not gain any nutrition if no damage was dealt? That method would keep the unique immunity to slimes IPCs have, as opposed to removing it and making them more similar to other species

I did consider it after I had already coded it, though I would rather not touch slime code again I could change it to no nutrition instead.

File: code/modules/mob/living/simple_animal/slime/slime_life.dm [hunk 3]
```diff
@@ -174,10 +175,20 @@
...
 		return
 
+// This is where damage dealt by slime feeding is handled.
 	if(iscarbon(M))
 		var/mob/living/carbon/C = M
-		C.adjustCloneLoss(rand(2, 4))
-		C.adjustToxLoss(rand(1, 2))
+		var/mob/living/carbon/human/D = M // I don't want to accidentally break feeding on Xenos or something.
+		if(C.dna) // Ensures there is DNA for the Synthetic check
+			if(C.dna.species.reagent_tag & PROCESS_SYN)
```
[the remark this answers is not in the document]
Comment (WenlockTheBritishHobo) [reply code/modules/mob/living/simple_animal/slime/slime_life.dm:183]: I had forgotten to test Vox but I can probably change the check so it's if the species can't process organic reagents instead.

**Comment by Onyxarias** [comment]
I'd rather have them continue to be immune to damage, but also have the ability for slimes to get nutrition from them removed. Personal preference, I want to see what the rest of Design thinks, but yeah.

**Comment by WenlockTheBritishHobo** [comment]
> I'd rather have them continue to be immune to damage, but also have the ability for slimes to get nutrition from them removed. Personal preference, I want to see what the rest of Design thinks, but yeah.

I forgot to reply last week so, gotcha. It would be easier to do it now since I actually have a decent idea of where that would be and how to do it.

File: code/modules/mob/living/simple_animal/slime/slime_life.dm [hunk 4]
```diff
@@ -174,10 +175,20 @@
...
+		if(C.dna) // Ensures there is DNA for the Synthetic check
+			if(!(C.dna.species.reagent_tag & PROCESS_ORG))
+				D.adjustBrainLoss(rand(2, 4)) // The IPC equivalent of Clone damage would be Brain damage.
+				D.adjustFireLoss(rand(1, 2), robotic = TRUE) // Poison can make you numb and feel on fire, also IPCs can't take Toxin damage.
+			else // Ensure slimes deal organic type damage to organics.
+				C.adjustCloneLoss(rand(2, 4))
+				C.adjustToxLoss(rand(1, 2))
+		else
+			C.adjustCloneLoss(rand(2, 4))
+			C.adjustToxLoss(rand(1, 2))
```
Comment (CinnamonSnowball) [inline code/modules/mob/living/simple_animal/slime/slime_life.dm:191]:
```suggestion
		if(C.dna?.species.reagent_tag & PROCESS_ORG))
			D.adjustBrainLoss(rand(2, 4)) // The IPC equivalent of Clone damage would be Brain damage.
			D.adjustFireLoss(rand(1, 2), robotic = TRUE) // Poison can make you numb and feel on fire, also IPCs can't take Toxin damage.
		else
			C.adjustCloneLoss(rand(2, 4))
			C.adjustToxLoss(rand(1, 2))
```

**Comment by S34NW** [comment]
This PR has unfortunately failed to pass the relevant team vote, with 1 for and 2 against.
Some of the reasons for objection include (anonymised):

- Yeah, slimes should not be getting any nutrition at all from an IPC. They should still take damage however because that can result in funny combat abusing with wearing a slime to prevent melee hits from hitting you with no downside.
- Currently voting against. I'd prefer that IPCs still not take damage from slimes. If slime armor had been a problem up to now it'd be different, but having the slimes just not gain nutrition from IPCs wont suddenly make IPCs more likely to wear slimes as armor.


Please feel free to discuss this with the relevant team and see if you can rework it into something acceptable to them.

**Pull request closed without merging**